Issue #16 - Add a script to prefill title, breakage, platform sections

ksy36 opened the pull request:
This is a helper script that could be useful for bugs like https://bugzilla.mozilla.org/show_bug.cgi?id=390936 with lots of breakage reports.

It can be run the following way:

```
python3 -m venv env
source env/bin/activate
pip install -r requirements.txt
python3 ./scripts/generate.py --bug_id=390936
```
It does the following:
- Creates a yml file in the data directory
- Prefills:
      -    title
      -    breakage reports from webcompat and fenix repository
      -    platform links (the original bugzilla bug and any others from bugs.chromium.org or bugs.webkit.org)

It doesn't add bugs from `duplicates` field or other bugzilla issues that linked in the `see also` field yet, so this needs to be done manually for now.

denschub commented on README.md:
```diff
@@ -12,6 +12,20 @@ Please note that this is still an early **work in progress**, and everything in
 
 - [Criteria for the `severity` and `user_base_impact` fields](./docs/severity-and-impact.md).
 
+## Script to create yml file for a given bugzilla bug
```
Can we maybe move this into the `docs/` folder and link to it in README.md, like we did with the severity docs above? I fully assume there will be more documentation over time, and having it split into separate files makes documentation a bit easier to discover and maintain :)

ksy36 replied:
Sounds good!

denschub commented on tools/scripts/generate.py:
```diff
@@ -0,0 +1,113 @@
+                contents = yaml.safe_load(stream)
+                # Check the first item in references -> platform_issues list to see if there is a match
+                if "references" in contents and "platform_issues" in contents["references"]:
+                    if contents["references"]["platform_issues"][0] == BUGZILLA_URL + bug_id:
```
> which of course won't flag the issue until you run the check, maybe not until pushing to GH

Side note, we could run the checks in a pre-push (or maybe even pre-commit) git hook. Once built, the current `kbcheck` runs in ~0.2s for me, so a git hook would be very reasonable!

ksy36 replied:
Yeah a git hook is a good idea. As for the check, we could probably keep it? In case someone uses the script and then fills in the rest of the sections and only after attempting to commit learns that an entry for that bug already exists (so the sooner a duplicate is detected, the better 🙂). I can sync the logic with the kbcheck.

ksy36 commented:
> Other than that, my only real worry is that we're adding tooling in a bunch of languages, and it feels like in the longer term we might want to standardise on a fewer tools so we aren't having to support the infrastructure for multiple languages (e.g. for this we should probably set up mypy for typechecking).

I agree, this is a valid point. I went with python as "quick and easy to start" solution. It's rather an experiment at this point, since I'm not sure whether the script is all that useful in it's current form for most of the bugs. I'm happy with picking one language in the longer term though :)

ksy36 commented:
> Having the type annotations here is good; I wonder if we should use them everywhere (i.e. also for parameters) and setup mypy?

I've added type annotations for parameters and a pre-commit hook for mypy.
Also changed the logic for finding duplicates and moved documentation to a separate file.

jgraham reviewed (approved):
OK, I'm going to apporve this because I think it's much better to land it than not, but I'd much prefer a CI job than a precommit hook, given that precommit hooks are hard to distribute and can interfere with workflows (e.g. make it harder to share in-progress changes).

ksy36 commented:
Thanks for the review :) I can add a CI job for mypy in a separate issue.
